components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4086,8 +4102,9 @@ public void removeApplication(Application application, String username) throws A
                                                                          String[] allowedDomains, String validityTime,
                                                                          String tokenScope, String groupingId,
                                                                          String jsonString,
-                                                                         String keyManagerName, String tenantDomain)
-            throws APIManagementException {
+                                                                         String keyManagerName, String tenantDomain,
+                                                                         boolean isImported)
```

Comment:
shall we rename this as `mapKeysIfKMAppCreationDisabled`?

Reply:
current usage of the argument is to map the keys, but kept the name generic so it can be used any time we need to identify if the flow is an import flow or normal key generation flow.

Reply:
Sure.. but "isImported" does not clearly say to me this is App import flow from CTL.. For me it's like we need to indicated the app is already imported.. not sure its only me though.. May be better to use "importMode" true/false ?

Reply:
How about `isImporting`? looks more conventional and explains what going on... WDYT? 😉

Reply:
Updated to `isImportMode`